pilot/pkg/xds/eds.go

```diff
@@ -362,6 +362,8 @@ func (eds *EdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w
 			}
 		}
 		builder := NewEndpointBuilder(clusterName, proxy, push)
+
+		// TODO(landow) we're using cached endpoints even if eps' networks change or eps are added from diff cluster
```

Comment:
What do you mean here? the cache is cleared in `AdsPushAll`. 

```
	if len(req.ConfigsUpdated) == 0 {
		s.cache.ClearAll()
	} else {
		// Otherwise, just clear the updated configs
		s.cache.Clear(req.ConfigsUpdated)
	}
```